sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -561,6 +564,367 @@ class FileBasedDataSourceSuite extends QueryTest
     }
   }
 
+  test("SPARK-31962 - when modifiedAfter specified " +
+      "with a past date") {
+    withTempDir { dir =>
+      withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC") {
+        val path = new Path(dir.getCanonicalPath)
+        val file = new File(dir, "file1.csv")
```

Comment:
Could you add tests for multiple file cases? Probably, I think you might be able to use `(new File("/tmp/file.csv")).setLastModified(xxx)` to control timestamp.

Reply:
Six more tests were added.